app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -39,7 +39,7 @@ class Api::V0::UserRolesController < Api::V0::ApiController
   # Filter role based on the permissions of the current user.
   private def can_current_user_access(role)
     group = role.group
-    !group.is_hidden || current_user&.has_permission?(:can_edit_groups, group.id)
+    !group.is_hidden || current_user&.has_permission?(:can_read_groups, group.id)
```

Comment:
Without any context, this seems like a pretty crazy change: What does "access" (from the method name `can_user_access`) mean in this context?
And why is it suddenly enough to have read-only permissions when previously you needed write permissions?

Reply:
I've changed this a bit in https://github.com/thewca/worldcubeassociation.org/pull/9378 and now this method is living in user_role.rb with the name `can_user_read?`, this should make it more clear.

This method is only used for read API. The reason why `can_edit_groups` is changed to `can_read_groups` is that, earlier both were having same value - ie, a user can edit a group if they can read the hidden group. But for banned competitors it's different - eg: WEC have access to read the hidden group, but not write. So created new permission.

Reply:
Okay, makes sense. Thanks for the explanation!